src/notifications/pollNotifications.ts

```diff
@@ -43,6 +57,34 @@ export async function doPollNotifications(
   );
 }
 
+async function handleConnectionHealthStatus(
+  context: vscode.ExtensionContext,
+  state: State
+): Promise<void> {
+  if (shouldShowConnectionLostNotification(state)) {
+    lastConnectionLostNotificationTime = new Date();
+    await handleNotification(createConnectionLostNotification(), context);
+  }
+}
+
+function createConnectionLostNotification() {
+  return {
+    id: `${CONNECTION_LOST_NOTIFICATION_ID_PREFIX}_${Date.now()}`,
+    ...CONNECTION_LOST_NOTIFICATION_PROPS,
+  };
+}
+
+function shouldShowConnectionLostNotification(state?: State | null) {
+  return (
+    state?.cloud_connection_health_status === "Failed" &&
+    (!lastConnectionLostNotificationTime ||
+      !isInTheLastHours(
+        lastConnectionLostNotificationTime,
+        INTERVAL_BETWEEN_CONNECTION_LOST_NOTIFICATIONS_HOURS
+      ))
+  );
+}
+
```

Comment:
can be moved to `src/notifications/connectionHealthNotification.ts` - just export `handleConnectionHealthStatus` from there.